qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -601,27 +601,29 @@ def _decomposer_2q_from_target(self, target, qubits, approximation_degree):
         # available instructions on this qubit pair, and their associated property.
         available_2q_basis = {}
         available_2q_props = {}
-        try:
-            keys = target.operation_names_for_qargs(qubits_tuple)
```

Comment:
I don't think this is correct because it is moving `target.operation_names_for_qargs()` outside of the `try`. But this line itself [can raise a KeyError.](https://github.com/Qiskit/qiskit-terra/blob/2f03a6ba720d27989a3c6ff74ede0b31733fa684/qiskit/transpiler/target.py#L681).
I think the reason my try blocks are large is that the Target API can raise KeyError on both `operation_names_for_qargs` and `operation_from_name` so I need both of these to be within a try.

Reply:
in the tests, the actual exception is raising in the `target[key][qubits_tuple]`. I feel that multiple exceptions is even a better reason for smaller blocks.

Reply:
Have another look, I tried again.